pyro/infer/autoguide/guides.py

```diff
@@ -306,18 +323,33 @@ def my_init_fn(site):
     :param callable model: A Pyro model.
     :param callable init_loc_fn: A per-site initialization function.
         See :ref:`autoguide-initialization` section for available functions.
+    :param callable create_plates: An optional function inputing the same
+        ``*args,**kwargs`` as ``model()`` and returning a :class:`pyro.plate`
+        or iterable of plates. Plates not returned will be created
+        automatically as usual. This is useful for data subsampling.
     """
-    def __init__(self, model, init_loc_fn=init_to_median):
+    def __init__(self, model, init_loc_fn=init_to_median, *,
+                 create_plates=None):
         self.init_loc_fn = init_loc_fn
         model = InitMessenger(self.init_loc_fn)(model)
-        super().__init__(model)
+        super().__init__(model, create_plates=create_plates)
 
     def _setup_prototype(self, *args, **kwargs):
         super()._setup_prototype(*args, **kwargs)
 
         # Initialize guide params
         for name, site in self.prototype_trace.iter_stochastic_nodes():
-            value = PyroParam(site["value"].detach(), constraint=site["fn"].support)
+            value = site["value"].detach()
+            event_dim = site["fn"].event_dim
+
+            # If subsampling, repeat init_value to full size.
+            for frame in site["cond_indep_stack"]:
+                full_size = getattr(frame, "full_size", frame.size)
+                if full_size != frame.size:
+                    dim = frame.dim - event_dim
+                    value = periodic_repeat(value, full_size, dim).contiguous()
```

Comment:
This logic was adapted from `EasyGuide.map_estimate()` which already supports subsampling.
https://github.com/pyro-ppl/pyro/blob/ac8b7e72916478783a39482dab4dbf71a9325127/pyro/contrib/easyguide/easyguide.py#L143-L153